blaze/compute/sql.py

```diff
@@ -176,6 +176,12 @@ def compute_up(t, s, **kwargs):
     return getattr(t, 'op', getattr(safuncs, sym, getattr(sa.func, sym)))(s)
 
 
+@dispatch(Selection, sa.sql.ColumnElement)
+def compute_up(expr, data, scope=None, **kwargs):
+    predicate = compute(expr.predicate, data, post_compute=False)
+    return sa.select([data]).where(predicate)
```

Comment:
i'm -1 on using that function as it hides a subtle aspect of sqlalchemy that can sometimes cause weird issues if you're not very careful about nesting selects